### PR TITLE
Gelu and leaky relu ops for mps

### PR DESCRIPTION
**Pull request by weiyanlin117**
unit test passed
Screenshot 2023-07-24 at 2 08 23 PM
Screenshot 2023-07-24 at 2 04 07 PM

### CODE REVIEW

File: lib/nnc/cmd/gelu/mps/ccv_nnc_gelu_mps.m [hunk 1]
```diff
@@ -53,6 +53,114 @@ static int _ccv_nnc_gelu_forw(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint
...
+  erfTensor = [mpsGraph erfWithTensor:erfTensor name:nil];
+  erfTensor = [mpsGraph additionWithPrimaryTensor:erfTensor secondaryTensor:onef name:nil];
+  erfTensor = [mpsGraph multiplicationWithPrimaryTensor:erfTensor secondaryTensor:halff name:nil];
+
+  return erfTensor;
+}
+
+static int _ccv_nnc_gelu_back(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint, const int flags, ccv_nnc_tensor_t* const* const inputs, const int input_size, ccv_nnc_tensor_t* const* const outputs, const int output_size, ccv_nnc_stream_context_t* const stream_context)
+{
+	assert(input_size == 3);
```
Comment (liuliu) [inline lib/nnc/cmd/gelu/mps/ccv_nnc_gelu_mps.m]: If you only use 0, 1, it probably should be `assert(input_size >= 2);`

File: lib/nnc/cmd/gelu/mps/ccv_nnc_gelu_mps.m [hunk 2]
```diff
@@ -53,6 +53,114 @@ static int _ccv_nnc_gelu_forw(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint
...
+			MPSGraphShapedType* mps_b_shape = ccv_nnc_mps_graph_tensor_input_shape(b, b->info.dim, b->stride);
+			[inputShapedTypes addObject:mps_b_shape];
+			MPSGraphTensor* inputTensor = mps_b;
+			MPSGraphTensor* gradTensor = mps_g;
+			MPSDataType dataType = mps_b.dataType;
+			MPSGraphTensor* mps_h;
+			if (cmd.info.gelu.tanh) {
+				 float kBeta = 0.797884560802865355 * (0.5f);
+				 float kKappa = 0.044715f;
+				MPSGraphTensor* betaf = [graph constantWithScalar:kBeta shape:@[ @1 ] dataType:dataType];
```
Comment (liuliu) [inline lib/nnc/cmd/gelu/mps/ccv_nnc_gelu_mps.m:102]: You don't need shape: https://developer.apple.com/documentation/metalperformanceshadersgraph/mpsgraph/3686986-constantwithscalar?language=objc

File: lib/nnc/cmd/gelu/mps/ccv_nnc_gelu_mps.m [hunk 3]
```diff
@@ -53,6 +53,114 @@ static int _ccv_nnc_gelu_forw(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint
...
+			MPSGraphTensor* mps_b = ccv_nnc_mps_graph_tensor_input(graph, b, b->info.dim, b->stride, &mps_input_b);
+			[inputTensors addObject:mps_input_b];
+			MPSGraphShapedType* mps_b_shape = ccv_nnc_mps_graph_tensor_input_shape(b, b->info.dim, b->stride);
+			[inputShapedTypes addObject:mps_b_shape];
+			MPSGraphTensor* inputTensor = mps_b;
+			MPSGraphTensor* gradTensor = mps_g;
+			MPSDataType dataType = mps_b.dataType;
+			MPSGraphTensor* mps_h;
+			if (cmd.info.gelu.tanh) {
+				 float kBeta = 0.797884560802865355 * (0.5f);
```
Comment (liuliu) [inline lib/nnc/cmd/gelu/mps/ccv_nnc_gelu_mps.m]: minor format nit. There are some whitespaces.

**Review by liuliu — APPROVED** [review]

**Comment by liuliu** [comment]
might want to fix the conflict by regenerate ...

**Comment by weiyanlin117** [comment]
test again after resolve conflicts and api updates 
Screenshot 2023-07-24 at 4 48 23 PM

**Pull request merged**